### ISSUE TITLE
Try reducing CI runtime

### ISSUE THREAD

**Issue by ethanfrey** [iov-one/iov-core#737]
`scripts/travis.sh` makes a number of checks and enforces at CI time what would otherwise be costly errors down the line. However, it is quite slow to give feedback and I think we can make it a bit quicker.

Some suggestions:

* We run `yarn lint` and then `yarn build`, which also does linting as a prebuild step. Let's just remove the explicit `yarn lint` step.
  * In [a sample build](https://travis-ci.com/iov-one/iov-core/jobs/173554099) `yarn lint` was 65s, `yarn build` was 129s. This would save a minute off CI
* `yarn docs` takes 90s and the only use is publishing them from master branch when pr is merged. There is already an "if statement" for this publishing condition, could we just build them on demand?

**Comment by webmaster128** [comment]
1. `yarn build` does not lint anymore (https://github.com/iov-one/iov-core/pull/597)
2. This step is not for building docs but for testing the docs building process works. Every code line can potentially break the typedoc build. Every typescript upgrade breaks the typedoc build.

Personally my strategy is push-and-forget, work on something else, come back – which works nice for me. But I can understand the annoyance when you want to get something done.

**Comment by webmaster128** [comment]
What we can do however is extracing lint and docs into their own build matrix job. So we get 4 jobs that can run in parallel:

1. Linux: MODE=lint
2. Linux: MODE=docs
3. Linux: MODE=test
4. Mac: MODE=test

This saves 2 1/2 minutes on the Linux job, probably even more on the Mac job (since Machine is slower), leads to smaller tasks that are restared on false positive and can detect multiple problems in one CI run at once.

**Comment by ethanfrey** [comment]
Nice solution. Maybe we can pull the node and browser tests in parallel under linux and make this even faster.

I don't see this as urgent, and I usually keep coding on stuff, but > 10 minute response time usually leads to me losing focus before I get a CI result... if we can target say 5-8 minute CI response, that would be great. But maybe we need a better Travis CI plan as well (pay for more power).

Not urgent, no need to work yet, but a "nice to have" sometime

**Comment by webmaster128** [comment]
#743 is a good start for a more parallalized build system. As I see the reduction of build time and stability of the CI as a continous process, I'm closing this in favour of smaller actionable tickets.
